impersonate_login/models/mail_message.py

```diff
@@ -0,0 +1,51 @@
+# Copyright (C) 2024 Akretion (<http://www.akretion.com>).
+# @author Kévin Roche <kevin.roche@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.http import request
+
+
+class Message(models.Model):
+    _inherit = "mail.message"
+
+    impersonated_author_id = fields.Many2one(
+        comodel_name="res.partner",
+        compute="_compute_impersonated_author_id",
+        store=True,
+    )
+
+    body = fields.Html(
+        compute="_compute_message_body",
+        store=True,
+    )
+
+    @api.depends("author_id")
+    def _compute_impersonated_author_id(self):
```

Comment:
This works for notes, but tracking messages still have the impersonated author:

![image](https://github.com/user-attachments/assets/deb7551a-f90c-4191-94e0-a78382c61633)

Actually, maybe the create_uid overwrite does not work. I am logged in as demo, impersonating a user called demo2. I created a crm.lead, but demo2 is still mentioned everywhere:

![image](https://github.com/user-attachments/assets/cca5a1f8-c9b3-4662-ad1b-20ac46e1853f)

Reply:
The tracking messages should be fixed now, would you check?

Regarding the create_uid, it seems to me correct that demo2 is mentioned there. Am I missing something?

Reply:
isn't it the case that the code attempts to inject the original user as the create_uid? Demo2 is the impersonated user.

Reply:
The note message now also has the create_uid and author_id of the impersonated user. I would expect both messages to have the create_uid of the original user.

![image](https://github.com/user-attachments/assets/d0149e51-f788-4876-9ef6-3bfda021722a)

Reply:
Ok sorry, I misunderstood your initial request. Reverted.

The note message is telling us: I'm Marc Demo (original user) and I'm impersonating Demo2 (impersonated user).
Do you mean that, when creating the crm.lead, the create_uid should say I'm Marc Demo (not Demo2)? Otherwise it seems correct to me.

Reply:
OK, I see the problem. Working on it.

Reply:
@StefanRijnhart 
With this commit https://github.com/OCA/server-auth/pull/680/commits/77f85f03eb504b319f500d2963cbdfe8bf4ccddb the create_uid should be solved. I was trying to do the same with the write_uid but I didn't manage to fix it.

Reply:
Indeed, it is not possible to override the create_uid in the write values, but it is possible to overwrite it using
```
    def write(self, vals):
        """Overwrite the write_uid with the impersonating user"""
        res = super().write(vals)
        if (
            request
            and request.session.impersonate_from_uid
            and "write_uid" in self._fields
        ):
            self._fields["write_uid"].write(self, request.session.impersonate_from_uid)
        return res
```

Reply:
It Works! Thank you very much!